### PR TITLE
Fix `sub_searches` serialization bug

### PR DESCRIPTION
**Pull request by jdconrad**
We have allowed hybrid search since 8.4. This means the internal changes for `sub_searches` must be able to write a compound query as early as 8.4, but currently we only do that back to 8.8. This change fixes that issue.

Closes #https://github.com/elastic/elasticsearch/issues/97144

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java [hunk 1]
```diff
@@ -292,7 +292,7 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeOptionalNamedWriteable(postQueryBuilder);
         if (out.getTransportVersion().onOrAfter(TransportVersion.V_8_500_013)) {
             out.writeList(subSearchSourceBuilders);
-        } else if (out.getTransportVersion().before(TransportVersion.V_8_8_0) && subSearchSourceBuilders.size() >= 2) {
+        } else if (out.getTransportVersion().before(TransportVersion.V_8_4_0) && subSearchSourceBuilders.size() >= 2) {
```
Comment (benwtrent) [inline server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java:295]: this only works if one of the subsearches is kNN and one is a query. Two queries breaks here right?

Comment (jdconrad) [reply server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java:295]: That is correct. However, I don't like the fragility of doing an `instanceof` check to ensure that one of these is a kNN query. Versions prior to kNN should break anyway when de-serializing the knn docs and score query. I view this as a courtesy to the user more than an actual exception as incorrect input here breaks the implicit contract of not using new features during upgrades that are unsupported.

**Comment by jdconrad** [comment]
@elasticsearchmachine run elasticsearch-ci/bwc

**Review by benwtrent — APPROVED** [review]

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Comment by benwtrent** [comment]
@elasticmachine elasticsearch-ci/full-bwc

**Comment by benwtrent** [comment]
@elasticmachine run elasticsearch-ci/full-bwc

**Comment by jdconrad** [comment]
@benwtrent Thanks for the review and adding full-bwc!

**Pull request merged**